uplink/eestream/unsafe_rs.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package eestream
+
+import (
+	"github.com/vivint/infectious"
+)
+
+type unsafeRSScheme struct {
+	fc               *infectious.FEC
+	erasureShareSize int
+}
+
+// NewUnsafeRSScheme returns a Reed-Solomon-based ErasureScheme without error correction.
+func NewUnsafeRSScheme(fc *infectious.FEC, erasureShareSize int) ErasureScheme {
+	return &unsafeRSScheme{fc: fc, erasureShareSize: erasureShareSize}
+}
+
+func (s *unsafeRSScheme) EncodeSingle(input, output []byte, num int) (err error) {
+	return s.fc.EncodeSingle(input, output, num)
+}
+
+func (s *unsafeRSScheme) Encode(input []byte, output func(num int, data []byte)) (
+	err error) {
+	return s.fc.Encode(input, func(s infectious.Share) {
+		output(s.Number, s.Data)
+	})
+}
+
+func (s *unsafeRSScheme) Decode(out []byte, in map[int][]byte) ([]byte, error) {
+	shares := make([]infectious.Share, 0, len(in))
```

Comment:
I think using lenght not capacity should be more optimal:
```
	shares := make([]infectious.Share, len(in))
	for num, data := range in {
		shares[num] = infectious.Share{Number: num, Data: data}
	}
```

Reply:
We can't do that since `num` in the map isn't the index but the piece number

Reply:
ah yes, my mistake. It still can be done with additional index but I think it won't give so much improvement.